### ISSUE TITLE
If Column value in first record is null, column type is erroneously set to string

### ISSUE THREAD

**Issue by Chirishman** [RamblingCookieMonster/PowerShell#26]
I'm working with `Out-DataTable` and AD and when I attempt to pass AD User objects through `Out-DataTable` it sets the column type of AccountExpirationDate to String because the value is null on the first record.

**Comment by RamblingCookieMonster** [comment]
Haha, good catch, looks like [I added a note](https://github.com/RamblingCookieMonster/PowerShell/blob/master/Out-DataTable.ps1#L117) when tweaking the original from Chad!

How would you expect this to work?  I suppose we could use something that [looks at all property types for all objects](https://github.com/RamblingCookieMonster/PowerShell/blob/master/Get-PropertyType.ps1), and if it sees things like null and string, it picks string (same for other types)?

I'd be up for a PR - if it could be done without too much of a mess, could have it behave as is by default, add a `ScanAllPropertyValues` or some similar switch that implements the more thorough, but slower code

Cheers!

**Comment by Chirishman** [comment]
Yeah, I've been thinking about the best way to solve it because it messes things up when I use it in conjunction with `Invoke-SQLiteBulkCopy`. For some reason I don't get an error on insert but I do on select when my select includes a column where a bad/conflicting typing has been created due to this bug.

**Comment by LUCScape** [comment]
Could offer an optional paramenter to input a list of data types of the columns?
